File: car.java
public class Car {
  private int mpg;
  private int gallons;
  private int odometerReading;
  private ArrayList<string> acidTrips;

  Car(mpg) {
    this.mpg = mpg;
    gallons = 0;
    odometerReading = 0;
    acidTrips = new ArrayList<string>();
  }

  public void fill(gallons) {
    this.gallons += gallons;
  }

  public void drive(miles) {
    gallons -= (miles/mpg);
    odometerReading += miles;
    acidTrips.add(miles + " miles");
  }

  public int odometer() {
    return odometerReading;
  }

  public ArrayList<string> trips() {
    return acidTrips;
  }
}

File: car.js
function Car(mpg) {
  var self = this;
  this.gallons = 0;
  this.odometerReading = 0;
  this.acidTrips = [];

  this.fill = function(gallons) {
    self.gallons += gallons;
  }

  this.drive = function (miles) {
    self.gallons -= (miles/mpg);
    self.odometerReading += miles;
    self.acidTrips.push(miles + ' miles');
  }

  this.odometer = function () {
    return self.odometerReading;
  }

  this.trips = function () {
    return self.acidTrips;
  }
};

module.exports = Car;
